crypto/keys/ed25519/ed25519.go

```diff
@@ -0,0 +1,232 @@
+package ed25519
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"io"
+
+	"github.com/tendermint/tendermint/crypto"
+	tmed25519 "github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"golang.org/x/crypto/ed25519"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+)
+
+//-------------------------------------
+
+const (
+	PrivKeyName = "cosmos/PrivKeyEd25519"
+	PubKeyName  = "cosmos/PubKeyEd25519"
+	// PubKeySize is is the size, in bytes, of public keys as used in this package.
+	PubKeySize = 32
+	// PrivateKeySize is the size, in bytes, of private keys as used in this package.
+	PrivateKeySize = 64
+	// Size of an Edwards25519 signature. Namely the size of a compressed
+	// Edwards25519 point, and a field element. Both of which are 32 bytes.
+	SignatureSize = 64
+	// SeedSize is the size, in bytes, of private key seeds. These are the
+	// private key representations used by RFC 8032.
+	SeedSize = 32
+
+	keyType = "ed25519"
+)
+
```

Comment:
How about using type alias or embed  stdlib `crypto/ed25519` PubKey and PrivateKey? With that we would avoid many convertions. Eg: if we update the struct to:

```
type PubKey struct {
-	Key []byte `protobuf:"bytes,1,opt,name=key,proto3" json:"key,omitempty"`
+	Key ed25519.PublicKey `protobuf:"bytes,1,opt,name=key,proto3" json:"key,omitempty"`
}
```
(same for PrivKey)

Code would be much simpler and faster.
The only cons is that we need to manually (or with `go:generate`) tweak the protobuf generated code. But maybe there is a way to instrument gogoproto which Go type it should use?

Reply:
`(gogoproto.casttype)` could be used but not sure to see the benefits, it would only avoid some conversions in this particular file.

Reply:
I can experiment with this in separate PR to not block this work.

Each `ed25519` operations will require conversion on each call.

Reply:
Created an issue for that: https://github.com/cosmos/cosmos-sdk/issues/7382